### DISCUSSION TITLE
Responsive UI Questions Concerning SLINT

### DISCUSSION THREAD

**Discussion by PackMeister** [slint-ui/slint#5810] [Q&A]
We're hoping to build an offline-first desktop prototype using SLINT and have following questions:

First to be clear, we mean by responsive a UI that adapts its size to either some default setting or by user dynamically dragging the UI window to make the UI's main window larger or smaller. 

Q1: Is UI responsiveness "baked into" SLINT designs built with SLINT's UI builder or must responsiveness be custom programmed for each and every UI.  Put another way, are there SLINT widgets or 'containers' that if used can provide (or inherit) UI responsiveness?

Q2: Do SLINT UI's require fixed dimensions?

Q3: If responsive desktop UIs are possible with SLINT are there examples with code showing how this can be accomplished?

**Comment by ogoffart** [comment]
The recommended way is to use layouts (HorizontalLayout, VerticalLayout, GridLayout or the *Box equivalent.)
And this is responsive.

So no fixed dimensions required.
I think most examples are responsive. The gallery, the Todo, the energy monitor, are just some examples

**Reply by PackMeister** [reply]
Thanks for your suggestions.

It looks like either the Fluent Design Gallery Demo or the Material Design Gallery Demo will be sufficient to explore the features we're seeking for our own SLINT prototype.

Though we plan to use Rust for our application logic for the time being we'd like to explore the Gallery Demos in Visual Studio Code using **only SLINT's own DSL** and **without Rust** (or any other language)......to test the idea of developing our UI completely independent of our application logic and its programming language.

In short, we'd like to readily explore SLINT's online demos of Fluent Design Gallery and Material Design Gallery locally (and dynamically) in VSC rather than with SLINT Pad.

Is this possible in VSC?

Are zip or git repos available for the above two demos that we can download and explore locally?

We're still Rust novices and would like to focus on learning SLINT by modifying the above demos and not get prematurely distracted or delayed with Rust-centric development practices whilst learning SLINT's DSL.

**Reply by tronical** [reply]
I think what you could do is the following:

1. Install VS Code
2. Install the Slint VS Code Extension
3. Clone the Slint git repository to a local directory
4. Open the `examples/gallery` folder in VS Code
5. Open `gallery.slint` in VS Code
6. Click `Show Preview` on the main window component
7. Enjoy the gallery in Slint Live-Preview running locally. You can change the sources, and you can change the style.

**Reply by PackMeister** [reply]
Thanks! 

That worked perfectly!
